### DISCUSSION TITLE
When viewing file changes, show changes since last view rather than all changes

### DISCUSSION THREAD

**Discussion by DimitriosLisenko** [community/community#7645] [Pull Requests]
# Description
During a PR, you can mark a file as viewed. However, if the file changes again while the PR is still open, *all* changes to that file are shown again, and a "Changed since last view" tag is added to the file. It would be much more useful to instead show changes *since the last view* rather than all changes to that file all over again.

I suppose this can be done by saving the git hash at the point that the user clicks "Viewed" per file, and then letting the user choose between seeing all changes vs seeing changes since last view for files.

# Example
1. Assume existing File X on branch A:
```
Hello, world!
```

2. Create and push branch B, which edits file X to:
```
Hello, world!
Toast is good!
```

3. Create PR from branch B to branch A - UI shows the following:
```
Hello, world!
Toast is good! # green because this line was added
```

4. Mark file as viewed in the PR

5. Edit file X in branch B to the following and push:
```
Hello, world!
Toast is good!
Buy a toaster now!
```

6. Current implementation will put a "Changed since last view" tag next to the file, and show the following:
```
Hello, world!
Toast is good! # green because this line was added
Buy a toaster now! # green because this line was added
```

7. Suggested change would show the following:
```
Hello, world!
Toast is good!
Buy a toaster now! # green because only this line was added since the last view
```

**Comment by IDrumsey** [comment]
I agree, or at least an option to view changes since last view. My issue relating to this is when I add my own review to my own PR and then add some changes. Usually if it's someone else's PR and I request changes and they fix them, there's a button that says "view changes since last review" which allows me to skip the stuff I've already reviewed. This would be a huge time saver.

**Comment by DimitriosLisenko** [comment]
I do believe that since I created this ticket this has actually been implemented.

![image](https://user-images.githubusercontent.com/6618354/228609770-5c775231-88ed-4465-8014-25fae9e8985a.png)
Clicking on "Changes from all commits" reveals a dropdown which now has the following option:
![image](https://user-images.githubusercontent.com/6618354/228609938-61679243-9641-40fd-9473-a61d9e85a58e.png)

It's not quite "changes since last view", but I've been quite happy with creating a review to mark completion, and then reviewing any changes since the last review using this option.

@IDrumsey admittedly I haven't tried this on my own PR - if that option is hidden for your own review, then I agree it would be useful to have as well.

**Reply by IDrumsey** [reply]
Oh, that is fantastic. It looks like I am able to view the changes since my own last review. Thanks for making me aware of this!

**Reply by ananda-cmd** [reply]
Is there a way to make that the default setting? Or does this option need to be selected every time a review is opened?

**Reply by edwardpassagi** [reply]
thank you, this is exactly what i was looking for!

**Comment by Thoralf-M** [comment]
Annoying thing with the changes from the new commits only is that you don't see them in the context with the other changes of the PR.
It would be nice if "Changes since last view" were simply displayed in a different color than the changes from previous commits.

```diff
Hello, world!
+ Toast is good! # green because this line was added
! Buy a toaster now!  # orange because only this line was added since the last view
```
(`+`, `!` are added to get the colors)

**Reply by IDrumsey** [reply]
That's a really good idea!

**Comment by AbhinavKoul** [comment]
A quick make-do can be if we are able to compare between commits inside a PR using `^` or `~` as mentioned [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparisons-across-commits)
